5_databases/week3/queries.sql

```diff
@@ -0,0 +1,104 @@
+----> Queries
+--> Meal
+SELECT *
+FROM meals
+
+
+INSERT INTO meals (title, description, location, meal_when, max_reservations, price, created_date)
+VALUES ('Shashlyk', 'Pork meat with onions on fire', 'Georgian restaurant', '2023-11-30 18:45:00', '14', '10.00', '2023-11-22')
+
+
+SELECT *
+FROM meals
+WHERE id = 1
+
+UPDATE meals 
+SET title = 'Kebab'
+WHERE id = 6;
+
+
+DELETE 
+FROM meals
+WHERE id = 1;
+
+--> Reservation
+
+SELECT *
+FROM reservations
+
+INSERT INTO reservations (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES (3, 2, '2023-11-28', '555-555-5555', 'Alex', 'alex@example.com');
+
+SELECT * 
+FROM reservations
+WHERE id = 1
+
+UPDATE reservations
+SET number_of_guests = 8 
+WHERE id = 1
+
+DELETE
+FROM reservations
+WHERE id = 1
+
+
+--> Review
+SELECT *
+FROM reviews
+
+
+INSERT INTO reviews (title, description, meal_id, stars, created_date)
+VALUES ('Great dish!', 'Really delicious dish, will definitely be back!', 2, 5, '2023-11-30');
+
+SELECT * 
+FROM reviews
+WHERE id = 1
+
+UPDATE reviews
+SET title = 'I have never eaten anything tastier'
+WHERE id = 1
+
+DELETE
+FROM reviews
+WHERE id = 1
+
+
+--> Additional queries
+
+SELECT * 
+FROM meals
+WHERE price < 15
+
+SELECT meals.* 
+FROM meals
+JOIN reservations ON meals.id = reservations.meal_id
+WHERE reservations.created_date < meals.meal_when
+AND reservations.number_of_guests < meals.max_reservations;
```

Comment:
Good approach. It is great that you only considered reservations that were created before the meal takes place and also considered the number of guest per reservation to see if there are any available spots left.

However, you query would show duplicated meals if it has multiple reservations. You should probably try to add up the total number of guests instead of just the guests in each reservation. How can you modify the below query to do that?

```sql
SELECT ???
FROM meal
LEFT JOIN reservation ON meal.id = reservation.meal_id
GROUP BY meal.id
HAVING meal.max_reservations > ???;
```